flask/app.py

```diff
@@ -825,14 +825,15 @@ def run(self, host=None, port=None, debug=None, **options):
                         information.
         """
         from werkzeug.serving import run_simple
-        if host is None:
-            host = '127.0.0.1'
-        if port is None:
-            server_name = self.config['SERVER_NAME']
-            if server_name and ':' in server_name:
-                port = int(server_name.rsplit(':', 1)[1])
-            else:
-                port = 5000
+
+        _host = '127.0.0.1'
+        _port = 5000
+        server_name = self.config.get("SERVER_NAME", None)
+        if server_name:
+            _host, _port = server_name.split(':', 1)
```

Comment:
https://docs.python.org/2/library/stdtypes.html#str.partition is somewhat useful for cases like this one